### DISCUSSION TITLE
QR Scanning Screens?

### DISCUSSION THREAD

**Discussion by patterueldev** [icerockdev/moko-widgets#307] [Ideas]
I haven't read into this too much. There's no proper documentation. But upon checking out the repository itself, I can't find any reference to QR scanners. I can only see **media** stuff like Media controllers that seem to use the camera. But I might need the AVFoundation or similar to android to display QR scanners.

I'm looking for something similar to flutter: https://pub.dev/packages/qr_code_scanner

I'm still trying to figure out how your library fully works. I don't know if it's possible to create platform-specific view controllers or activities by injecting some code to make it compatible with routing? This is in case a certain UI implementation is still not available, so I would like to [do it myself](https://i.imgflip.com/2u7l7u.png).

**Comment by Alex009** [comment]
hi @jpteruel095 
here no qr scanners out of box, but you can do it by:

1. custom widget 
- here sample of creation single widget https://github.com/icerockdev/moko-widgets/tree/master/widgets-collection

2. native screen
- https://github.com/icerockdev/moko-widgets/blob/master/sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/universal/ProfileScreen.kt
- https://github.com/icerockdev/moko-widgets/blob/master/sample/mpp-library/src/androidMain/kotlin/com/icerockdev/library/universal/PlatformProfileScreen.kt
- https://github.com/icerockdev/moko-widgets/blob/master/sample/mpp-library/src/iosMain/kotlin/com/icerockdev/library/universal/PlatformProfileScreen.kt
